### ISSUE TITLE
branch 4.039 fileset structure

### ISSUE THREAD

**Issue by torvista** [mc12345678/EasyPopulate-4.0#70]
Just trying out the latest and greatest branch, but find this complex structure confusing:
![2024-03-24 12_01_33-Multi Commander (x64)  v13 5 (build 2983)](https://github.com/mc12345678/EasyPopulate-4.0/assets/4391026/7303ae85-2e1e-43d7-8d55-ab3bc36c20c8)

Not an easy copy/merge!

I suspect some of these should be in zc_plugins...?

**Comment by mc12345678** [comment]
The sub-folder `EasyPopulateV4` is the directory to include in the `zc_plugins` directory. 

For a pre-ZC 1.5.7 site, just using the base `includes` and `admin` directories would do.

Regardless the installation method, the `htaccess4AdminTempFolder` may be helpful, but is not required. 

The temporary/storage directory though is still in relation to the store, not to specifically to the plugins directory. 

Development problems identified a few years ago is that while a version (e.g., `v4.0.39.ZC`) is intended to hold the code supporting it, the next/upcoming version needs to have some sort of fluidity. In this way some support can be maintained for older versions of Zen Cart in support of continuing to add functionality while still maintaining operation.

If you have some suggestions on folder structure, feel free to offer. 

My next thought is to create a `zc_plugins` sub-directory and locate all of the associated files within it. The next version would get a new sub-directory within the `zc_plugins/EasyPopulateV4/` sub-directory. I doubt still though that direct access of github will make it easier to figure out how to move forwards. And that the plugin as uploaded to the zen-cart.com site will have both methods of retrieval. Hopefully the instructions when edited will be able to make sense.

**Comment by torvista** [comment]
>The sub-folder EasyPopulateV4 is the directory to include in the zc_plugins directory.

Yes, I assumed that, but then there are duplicated directories in the root
GitHub\EasyPopulate-4.0\catalog
GitHub\EasyPopulate-4.0\Installer

>My next thought is to create a zc_plugins sub-directory and locate all of the associated files within it. 

That's what surprised me, that the relevant files are not already in that directory.

Incidentally, I did copy those files to the zc_plugins directory (and removed the ones in the main fileset) but EP does not appear in the list in the admin Plugin Manager.

**Comment by mc12345678** [comment]
> > The sub-folder EasyPopulateV4 is the directory to include in the zc_plugins directory.
> 
> Yes, I assumed that, but then there are duplicated directories in the root GitHub\EasyPopulate-4.0\catalog GitHub\EasyPopulate-4.0\Installer
> 
> > My next thought is to create a zc_plugins sub-directory and locate all of the associated files within it.
> 
> That's what surprised me, that the relevant files are not already in that directory.
> 
> Incidentally, I did copy those files to the zc_plugins directory (and removed the ones in the main fileset) but EP does not appear in the list in the admin Plugin Manager.

Understand, agree, and will seek to correct. In the interim, files in the "root" need to be placed in the appropriate location within the `zc_plugins` file structure (more to follow for those interested). Thought I had done that, which probably is true in my local copy. Unfortunately those new files do not yet display as part of the repo. They need to be committed to the repo and then after that I will likely relocate all `EasyPopulateV4` folder contents into a `zc_plugins` sub-directory.

**Comment by mc12345678** [comment]
Yup. As I expected, files were "loaded" to the directory; however, had not been committed to the repo.

Corrected by commit https://github.com/mc12345678/EasyPopulate-4.0/commit/c244faa303c18dbcf5e8c961e0e45c7d56a54646

Next to move the files into a "clearer" yet still somewhat obscure `zc_plugins` directory :).

**Comment by mc12345678** [comment]
`zc_plugins` directory generated and populated by commit: https://github.com/mc12345678/EasyPopulate-4.0/commit/191f05aeb6bef5b999ecf45015a2ddc45093ef09

**Comment by torvista** [comment]
Hmmm, there is still in the root:

Installer
catalog

which appear to be superfluous now.

And, if EP can be installed by the Plugin Manager, the admin and includes folders can also go?

And for me, any reference to ancient history like, pre-158.
Developer time is scarce and I don't agree with worrying about how todays Plugin copes with yesterday's ZC.
People should upgrade before adding more recent plugins.

**Comment by torvista** [comment]
I see there are some minor issues with the install/uninstall via Plugin Manager to resolve for ZC200, otherwise ok.
